### DISCUSSION TITLE
[Proposal]: Implicit Cancellation Token

### DISCUSSION THREAD

**Discussion by Elviszip** [dotnet/csharplang#7327] [Language Ideas]
# Implicit Cancellation Token

* [x] Proposed
* [ ] Prototype: Not Started
* [ ] Implementation: Not Started
* [ ] Specification: Not Started

## Summary

Allow methods to accept ImplicitCancellationToken as a parameter.

## Motivation
To reduce boilerplate of passing cancelation tokens

## Detailed design

```
public object DoSomething(parameter one..., ImplicitCancellationToken token)
{
  method(params...); // cancelationToken added by compiler (implicitely!)
}
```

## Drawbacks
[drawbacks]: #drawbacks

## Alternatives
[alternatives]: #alternatives

## Unresolved questions
[unresolved]: #unresolved-questions

**Comment by HaloFour** [comment]
See: #5989

**Comment by theunrepentantgeek** [comment]
This was one of my motivating examples for #1332
